UnitTests/GitCommands.Tests/Submodules/SubmoduleStatusProviderTests.cs

```diff
@@ -30,10 +33,23 @@ internal class SubmoduleStatusProviderTests
         private GitModule _repo3Module;
 
         private ISubmoduleStatusProvider _provider;
-
+        private bool _isInit = false;
+
+        /// <summary>
+        /// The repo creation should only be done once as it slows tests down considerably
+        /// Each test must therefore revert changes after the test
+        /// OneTimeSetUp cannot be used, as the setup must run after BeforeTest in CommonTestUtils\ConfigureJoinableTaskFactoryAttribute.cs
+        /// OneTimeTearDown can be used though
+        /// </summary>
         [SetUp]
         public void SetUp()
         {
+            if (_isInit)
+            {
+                return;
+            }
+
+            _isInit = true;
             _repo1 = new GitModuleTestHelper("repo1");
```

Comment:
Tests shouldn't concern themselves with resetting the state after execution. I was thinking of a change along the lines of `ReferenceRepoistory`:
```cs
            if (_repo1 is null)
            {
                _repo1 = new GitModuleTestHelper("repo1");
            }
            else
            {
                _repo1.Reset();
            }
```

Do you think we can re-write these tests to utilise `ReferenceRepository` instead?

Reply:
The repos would need not just to reset, but also undo commits. That would require that the initial commit have to be remembered, which will make a more complex setup. Also, a reset will increase the test time slightly.
The test cases cleaning up is much easier.

Reply:
A clean is needed too, in addition to reset and checkout.